app/src/main/java/io/appium/uiautomator2/utils/AXWindowHelpers.java

```diff
@@ -85,13 +87,20 @@ public static void refreshRootAXNode() {
      */
     public static AccessibilityNodeInfo[] getWindowRoots() {
         List<AccessibilityNodeInfo> ret = new ArrayList<>();
+
         /*
-         * TODO: MULTI_WINDOW is disabled, UIAutomatorViewer captures active window properties and
-         * end users always relay on UIAutomatorViewer while writing tests.
+         * TODO: MULTI_WINDOW is disabled by default, UIAutomatorViewer captures active window
+         * properties and end users always relay on UIAutomatorViewer while writing tests.
          * If we enable MULTI_WINDOW it effects end users.
          * https://code.google.com/p/android/issues/detail?id=207569
          */
-        if (CustomUiDevice.getInstance().getApiLevelActual() >= Build.VERSION_CODES.LOLLIPOP && MULTI_WINDOW) {
+
+        boolean multiWindowsEnabled = AppiumUIA2Driver
```

Comment:
this should be read from the setting

Reply:
Can you point me to an example of reading from the settings in the code? I saw that way at first

Reply:
https://github.com/appium/appium-uiautomator2-server/blob/f117326c6695a4dbb6382ecb6e77049df3874817/app/src/main/java/io/appium/uiautomator2/server/ServerInstrumentation.java#L196